src/scopes.js

```diff
@@ -56,3 +60,35 @@ exports.scopeMatch = function(scopePatterns, scopesets) {
     });
   });
 };
+
+/**
+ * Finds scope intersections between two scope sets.
+ */
+exports.scopeIntersection = (scopeset1, scopeset2) => {
+  validateScopePatterns(scopeset2);
+  validateScopePatterns(scopeset1);
+
```

Comment:
Consider using something like:
https://github.com/taskcluster/taskcluster-auth/blob/e7517fddeb0615ab80472be75919adf4c0f1b68a/src/dfa.js#L86-L88

I know the `scopeCompare` is a non-trivial, but it's well-tested..

This is basically a sorting here `abc*` comes before `abc`, which comes before `abcd...`.
Once sorted this way you can by doing a scan of both lists in parallel... As soon as you see `abc*` in `scopes1` then all the scopes in `scopes2` that starts with `abc` can be added to intersection.